### PR TITLE
TH-861: Start to use helsinki design system Footer component

### PR DESCRIPTION
**Pull request by nenonja**
## Description :sparkles:

~~Work-in-progress:~~
- [x] Revert categories back
- [x] Fix background colors

Unify footer by using helsinki design system's Footer Component instead of custom Footer:
https://app.abstract.com/projects/b362e050-6671-11e9-9013-21682b9a514b/branches/master/commits/fc7c623eabe6c5846590a8def22f52c8e9cf1f3e/files/72EAE992-755F-4ED8-8152-1F272C996F99/layers/CA33CFF9-2FBB-46CA-BBCC-755F5E383F84?collectionId=f7f4dedc-8bc9-4be7-b09c-99943a0d8aee&collectionLayerId=e36a6aa8-82cd-4b4a-9a3e-c025382630e5&mode=build&selected=1174534909-6C0BF8FF-A1FB-4E3C-A666-63CB00106394

## Issues :bug:
https://helsinkisolutionoffice.atlassian.net/browse/TH-861

### Fullscreen:

Screenshot 2020-12-10 at 6 01 59 PM

### Ipad:

Screenshot 2020-12-10 at 6 02 55 PM

### Mobile:

Screenshot 2020-12-10 at 6 02 16 PM

## Additional notes :spiral_notepad:
- ~~Should We add Social Media icons? Footer supports them.~~
- Helsinki logo picture is not changed to swedish (Helsingfors). It's limitation by HDS Footer component: https://helsinkicity.slack.com/archives/CHCV3KTHA/p1607537742116800
- ~~footer content does not support same max width as rest of the page, so it is narrower than rest of the page. Same problem is with Header too.~~ will be done in another ticket

### CODE REVIEW

File: src/domain/app/footer/footerCategories.module.scss [hunk 1]
```diff
@@ -0,0 +1,40 @@
+@import 'layout';
```
Comment (nenonja) [inline src/domain/app/footer/footerCategories.module.scss:1]: old `topFooter.module.css` with some changes but it seems that git fails to show changed rows after file rename.

File: src/domain/app/footer/FooterCategories.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import React, { FunctionComponent } from 'react';
```
Comment (nenonja) [inline src/domain/app/footer/FooterCategories.tsx:1]: old `TopFooter.tsx` with some changes but it seems that git fails to show changed rows after file rename.

File: src/domain/app/footer/footerCategories.module.scss [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  height: fit-content;
+  grid-template-columns: 1fr;
+  grid-gap: 1rem 3.25rem;
+
+  @include respond-above(sm) {
+    grid-template-columns: 1fr 1fr;
+  }
+
+  @include respond-above(md) {
+    grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
```
Comment (jorilindell) [inline src/domain/app/footer/footerCategories.module.scss:38]: Maybe `grid-template-columns: repeat(5, 1fr)` could be used to make it little bit more readable?

File: src/domain/app/footer/Footer.tsx [hunk 4]
```diff
@@ -1,15 +1,59 @@
...
 import React, { FunctionComponent } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
 
-import BottomFooter from './BottomFooter';
-import TopFooter from './TopFooter';
+import useLocale from '../../../hooks/useLocale';
+import { ROUTES } from '../constants';
+import FooterCategories from './FooterCategories';
+import styles from './footerCategories.module.scss';
```
[the remark this answers is not in the document]
Comment (nenonja) [reply src/domain/app/footer/Footer.tsx:9]: We should make eslint to complain / fix these

File: src/domain/app/footer/__tests__/Footer.test.tsx [hunk 5]
```diff
@@ -1,10 +1,17 @@
...
+  const component = renderer.create(
+    <MemoryRouter>
+      <Footer />
+    </MemoryRouter>
+  );
+  const tree = component.toJSON();
+  expect(tree).toMatchSnapshot();
 });
+
+export {};
```
[the remark this answers is not in the document]
Comment (nenonja) [reply src/domain/app/footer/__tests__/Footer.test.tsx:17]: This file should not have any changes, I will revert the changes.

File: src/domain/app/footer/__tests__/Footer.test.tsx [hunk 6]
```diff
@@ -1,10 +1,17 @@
...
-
-  expect(container.firstChild).toMatchSnapshot();
+test('Footer matches snapshot', () => {
+  const component = renderer.create(
+    <MemoryRouter>
+      <Footer />
+    </MemoryRouter>
+  );
+  const tree = component.toJSON();
+  expect(tree).toMatchSnapshot();
```
[the remark this answers is not in the document]
Comment (nenonja) [reply src/domain/app/footer/__tests__/Footer.test.tsx:14]: This file should not have any changes, I will revert the changes.

File: src/domain/app/footer/footerCategories.module.scss (hunk 3 above)
Comment (nenonja) [reply src/domain/app/footer/footerCategories.module.scss:38]: This was from original file which was renamed but I can fix this.

**Review by Sartonon — APPROVED** [review]
Great!!!!!

**Pull request merged**